cli/command/node/completion.go

```diff
@@ -0,0 +1,36 @@
+package node
+
+import (
+	"os"
+
+	"github.com/docker/cli/cli/command/completion"
+	"github.com/docker/docker/api/types"
+	"github.com/spf13/cobra"
+)
+
+// completeNodeNames offers completion for swarm node (host)names and optional IDs.
+// By default, only names are returned.
+// Set DOCKER_COMPLETION_SHOW_NODE_IDS=yes to also complete IDs.
+//
+// TODO(thaJeztah): add support for filters.
+func completeNodeNames(dockerCLI completion.APIClientProvider) completion.ValidArgsFn {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		list, err := dockerCLI.Client().NodeList(cmd.Context(), types.NodeListOptions{})
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		showIDs := os.Getenv("DOCKER_COMPLETION_SHOW_NODE_IDS") == "yes"
```

Comment:
To be consistent with other options, we should either:

```suggestion
		showIDs := os.Getenv("DOCKER_COMPLETION_SHOW_NODE_IDS") != ""
```

or parse as bool:
https://github.com/docker/cli/blob/f9ced58158d5e0b358052432244b483774a1983d/cli/hints/hints.go#L12-L19

Reply:
Also, can be moved out of the returned function, directly to `completeNodeNames`

Reply:
> To be consistent with other options, we should either:

Yeah, point was that the old script also allowed `"no"` to be set; https://github.com/docker/cli/blob/f9ced58158d5e0b358052432244b483774a1983d/contrib/completion/bash/docker#L41-L43


We could use ParseBool, but currently don't do so for the equivalent option for containers, so I (for now) picked the same here.